src/StringViews.jl

```diff
@@ -58,6 +59,7 @@ Base.unsafe_convert(::Type{Ptr{UInt8}}, s::DenseStringViewAndSub) = pointer(s)
 Base.unsafe_convert(::Type{Ptr{Int8}}, s::DenseStringViewAndSub) = convert(Ptr{Int8}, pointer(s))
 Base.cconvert(::Type{Ptr{UInt8}}, s::DenseStringViewAndSub) = s
 Base.cconvert(::Type{Ptr{Int8}}, s::DenseStringViewAndSub) = s
+Base.cconvert(::Type{Cstring}, s::DenseStringViewAndSub) = s.data
```

Comment:
Why not simply `s`?

Reply:
Yes, probably we need the [default conversion](https://github.com/JuliaLang/julia/blob/37082297452099ec262cd6100fe2aaa1717d7aa9/base/c.jl#L197-L198) here.

Reply:
I’m not convinced it is safe to assume that the data here is NUL-terminated. Have to go back and check what guarantees Julia provides